src/control/drpc/status.go

```diff
@@ -25,130 +25,145 @@
 
 package drpc
 
-// #include <gurt/errno.h>
+import "fmt"
+
+/*
+#cgo LDFLAGS: -lgurt
+
+#include <gurt/errno.h>
+*/
 import "C"
 
 // DaosStatus is a status code in the set defined by the DAOS data plane.
 type DaosStatus int32
 
+func (ds DaosStatus) Error() string {
+	// NB: Currently, d_errstr() just returns a string of the status
+	// name, e.g. -1007 -> "DER_NOSPACE". This is better than nothing,
+	// but hopefully d_errstr() will be extended to provide better strings
+	// similar to perror().
+	dErrStr := C.GoString(C.d_errstr(C.int(ds)))
+	return fmt.Sprintf("DAOS error (%d): %s", ds, dErrStr)
+}
+
 const (
 	// DaosSuccess indicates no error
 	DaosSuccess DaosStatus = 0
 	// DaosNoPermission indicates that access to a resource was denied
-	DaosNoPermission = -C.DER_NO_PERM
+	DaosNoPermission DaosStatus = -C.DER_NO_PERM
```

Comment:
For some reason I thought the type was implied by the first item in the list in these const lists. Is that only when using iota?

Reply:
Yeah, with iota you set the type in the first assignment and then it's carried through to the other implicit values.